docs/proposal-006-build-improvements.md

```diff
@@ -0,0 +1,55 @@
+# Design Proposal #006: Build Improvements
+
+## Current State
+
+At the start of the project we've picked CircleCI as it appears to be a default choice at Weaveworks.
+
+The original solution involved:
+- running `docker build` to create a build image
+- use build image to run tests produce final image
+
+Publishing build and final images hasn't been automated, but some users requested it.
+
+When we switched [to modules](https://github.com/weaveworks/eksctl/pull/917), we have increased the
+compexity of how build image worked. It's hard for a new person to understand it, e.g. for a new
+contributor.
+
+We currently build images in CI and run tests as part of that, we should only run tests as a primary
+task.
+
+Downside of current setup:
+
+- CircleCI requires delegation of access to GitHub
+  - for releases we use a token
+  - we don't have a way to automate workflows that require pushing to the repo
+- CircleCI cache 
+  - it is not very fast, it turns out pulling image with cached dependencies is as fast
+  - it is very specifig to CircleCI, not portable
```

Comment:
what parts are specific to circle ci?

Reply:
as above, this is under 'CircleCI cache'

Reply:
Ok, so what happens to the cache in the new proposed solution?

Reply:
It will be stored inside of the build image.

Reply:
Please mention that and resolve this comment.